luxonis_ml/nn_archive/config_building_blocks/base_models/output.py

```diff
@@ -49,3 +51,11 @@ def validate_layout(self) -> Self:
             raise ValueError("Layout and shape must have the same length.")
 
         return self
+
+    @model_validator(mode="after")
```

Comment:
What is the reson that for input we validate `before` and for output we validate `after`?

Reply:
Output's layout is optional, so we can wait and validate after. For input it's mandatory, so it has to be validated before in order to determine whether the layout was explicitly provided (no inference in that case), or not.